tsl/src/compression/compress_utils.c

```diff
@@ -218,6 +250,9 @@ compress_chunk_impl(Oid hypertable_relid, Oid chunk_relid)
 	hcache = ts_hypertable_cache_pin();
 	compresschunkcxt_init(&cxt, hcache, hypertable_relid, chunk_relid);
 
+	/* Perform an analyze on the chunk to get up-to-date stats before compressing */
+	preserve_uncompressed_chunk_stats(chunk_relid);
```

Comment:
Why do we need to preserve old stats? How is this useful? We need the compressed chunk stats anyway to make the right decisions for planning. (cc @svenklemm  Your thoughts?)

Reply:
The pg_statistics give us things like most common value and histograms which aren't available through the compressed chunk stats.  While perhaps not as useful as stats such as live_tuples, these are still useful for choosing optimal plans in some cases.